lightning/src/routing/scoring.rs

```diff
@@ -623,6 +623,35 @@ impl<G: Deref<Target = NetworkGraph>, L: Deref, T: Time> ProbabilisticScorerUsin
 		assert!(self.channel_liquidities.insert(short_channel_id, liquidity).is_none());
 		self
 	}
+
+	/// Dump the contents of this scorer into the configured logger.
+	///
+	/// Note that this writes roughly one line per channel for which we have a liquidity estimate,
+	/// which may be a substantial amount of log output.
+	pub fn debug_log_liquidity_stats(&self) {
+		let graph = self.network_graph.read_only();
+		for (scid, liq) in self.channel_liquidities.iter() {
+			if let Some(chan_debug) = graph.channels().get(scid) {
+				let log_direction = |dir: &Option<ChannelUpdateInfo>, source, target| {
```

Comment:
You should be able to drop the first parameter because the behavior is encapsulated in `DirectedChannelInfo`.